perception/autoware_ground_segmentation_cuda/src/cuda_scan_ground_segmentation/cuda_scan_ground_segmentation_filter.cu

```diff
@@ -0,0 +1,819 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/cuda_scan_ground_segmentation/cuda_scan_ground_segmentation_filter.hpp"
+
+#include <autoware/cuda_scan_ground_segmentation/cuda_common.hpp>
+#include <autoware/cuda_scan_ground_segmentation/cuda_utility.cuh>
+#include <cub/cub.cuh>
+
+#include <sensor_msgs/msg/point_field.hpp>
+
+#include <sys/time.h>
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <optional>
+
+namespace autoware::cuda_ground_segmentation
+{
+
+__device__ __forceinline__ float fastAtan2_0_2Pi(float y, float x)
+{
+  return fmodf(atan2(y, x) + 2.0f * M_PI, 2.0f * M_PI);
+}
+
+// Initialize the cell list
+__global__ void cellInit(Cell * __restrict__ cell_list, int max_num_cells)
+{
+  int index = threadIdx.x + blockIdx.x * blockDim.x;
+  int stride = blockDim.x * gridDim.x;
+  Cell new_cell;
+
+  for (int i = index; i < max_num_cells; i += stride) {
+    new_cell.gnd_height_min = 1e6f;
```

Comment:
nits: If `1e6f` means super large value, you can also use `::cuda::std::numeric_limits<float>::max()` or `::cuda::std::numeric_limits<float>::infinity()`

Reply:
I fixed here 0549db2d82cc1356d16ccbc4ea6053e7c7cd66dd